app/classes/textile.rb

```diff
@@ -207,25 +209,30 @@ def self.clear_textile_cache
   private
 
   NAME_LINK_PATTERN = /
-    (^|\W) (?:\**_+) ([^_]+) (?:_+\**) (?= (?:s|ish|like)? (?:\W|\Z) )
-  /x
-  OTHER_LINK_PATTERN = /
-   (^|\W) (?:_+) ([a-zA-Z]+) \s+ ([^_\s](?:[^_\n]+[^_\s])?) (?:_+) (?!\w)
+    (?<prefix>^|\W)
+    (?:\**_+)
+    (?<formatted_label>[^_]+)
+    (?:_+\**)
+    (?= (?:s|ish|like)?
+    (?:\W|\Z) )
```

Comment:
I would keep this zero-width, just to be safe.  I think you might be able to test this case with the string `_Agaricus_ _Amanita_` with just a single space between.  I suspect it might not "see" the second link unless you keep this "postfix" zero-width (`?=` instead of `?:`).